src/ASM/DomainDecomposition.C

```diff
@@ -27,109 +27,91 @@
 #include <numeric>
 
 
-//! \brief Iterator for matching nodes on edges/faces with a given orientation and index.
-class NodeIterator {
-public:
-  //! \brief Constructor.
-  //! \param pch Slave patch
-  //! \param orient Orientation of boundary on slave
-  //! \param lIdx Index of boundary on slave
-  //! \param basis Basis to use for boundary on slave
-  //! \param dim Dimension to iterate over
-  NodeIterator(const ASMbase* pch, int orient, int lIdx, int basis, int dim = 2)
-  {
-    if (dim == 0) {
-      nodes.resize(1);
-      return;
-    }
-    const ASMstruct* spch = dynamic_cast<const ASMstruct*>(pch);
-    const ASMunstruct* upch = dynamic_cast<const ASMunstruct*>(pch);
-
-    int nsd = pch->getNoSpaceDim();
-    if (upch) {
-      if (nsd == 3 && dim == 1) {
-        const ASM3D* pch3D = dynamic_cast<const ASM3D*>(pch);
-        nodes = pch3D->getEdge(lIdx, true, basis, -1);
+DomainDecomposition::
+OrientIterator::OrientIterator(const ASMbase* pch,
+                               int orient, int lIdx, int basis, int dim)
+{
+  if (dim == 0) {
+    nodes.resize(1);
+    return;
+  }
+  const ASMstruct* spch = dynamic_cast<const ASMstruct*>(pch);
+  const ASMunstruct* upch = dynamic_cast<const ASMunstruct*>(pch);
+
+  int nsd = pch->getNoSpaceDim();
+  if (upch) {
+    if (nsd == 3 && dim == 1) {
+      const ASM3D* pch3D = dynamic_cast<const ASM3D*>(pch);
+      nodes = pch3D->getEdge(lIdx, true, basis, -1);
+      std::iota(nodes.begin(), nodes.end(), 0);
```

Comment:
don't quite see the point with the `getEdge()` call here since you in the next line overwrite the result with `std::iota` is is just a complicated way to obtain the number of element nodes along the edge?

Reply:
yup. it's to get number of nodes. this is the nodes constructor, not the element constructor (old code). for unstructured patches we have to do the ordering inside the function that grabs the nodes so the iterator is always linear. instead of adding another method to get the number of nodes, this does the job.